sky/clouds/azure.py

```diff
@@ -249,13 +249,19 @@ def make_deploy_resources_variables(
         # This script will modify /etc/ssh/sshd_config and add a bash script
         # into .bashrc. The bash script will restart sshd if it has not been
         # restarted, identified by a file /tmp/__restarted is existing.
+        # We also stop jupyterhub and jupyter to avoid port conflict on 8081
+        # and 8888.
         # pylint: disable=line-too-long
         cloud_init_setup_commands = base64.b64encode(
             textwrap.dedent("""\
             #cloud-config
             runcmd:
               - sed -i 's/#Banner none/Banner none/' /etc/ssh/sshd_config
               - echo '\\nif [ ! -f "/tmp/__restarted" ]; then\\n  sudo systemctl restart ssh\\n  sleep 2\\n  touch /tmp/__restarted\\nfi' >> /home/azureuser/.bashrc
+              - systemctl stop jupyterhub
+              - systemctl disable jupyterhub
+              - systemctl stop jupyter
+              - systemctl disable jupyter
```

Comment:
Are these commands idempotent? What will happen if running `stop` after the jupyterhub is already `disabled`?

If we are not disabling the service before the service is started, I think it does not matter to have it in the cloud-init or the setup commands. It might be a bit better to have it in the setup commands to be more observable.

Reply:
Just confirmed, the service is started when cloud-init runcmd is executed...

Reply:
I'll change back to setup commands